### ISSUE TITLE
Package should not fail verification due to a dependency failure

### ISSUE THREAD

**Issue by matbech** [chocolatey/package-verifier#40]
The verification for a package should not fail if only the dependency (which may be exempted from the test) fail to install. As an example please see:
https://gist.github.com/choco-bot/4530543d9f68b702ddc3
In this particular case the installation of vcredist2015 fails. However, the vcredist2015 package has been exempted from verification and thus shouldn't make its dependent packages fail.

**Comment by ferventcoder** [comment]
I disagree.

If the package cannot be installed because of a failing dependency, that package is also broken. If a dependent package is broken, the dependent package needs fixed so that every package that depends on it doesn't break.

I would consider vcredist a temporary solution until we get a KB package in place under it.

**Comment by ferventcoder** [comment]
I would consider vcredist skipping verification a temporary thing until we get a KB package in place under it.

**Comment by gep13** [comment]
@ferventcoder if I read the issue correctly, the situation being described is...

If a package installation fails due to a package dependency failing, **and** that dependency has an **exemption** from testing, then the package being installed should also be exempt.

I tend to agree with this.

**Comment by ferventcoder** [comment]
Yes, I agree with what you are saying wholeheartedly @gep13 - I took this to mean that somehow we should make the verifier ignore the failing dependency.

**Comment by ferventcoder** [comment]
@matbech is what @gep13 was stating what you meant? Or did you mean that we should try to find some way to ignore a failing dependency?

**Comment by matbech** [comment]
@ferventcoder Yes, one case is the one @gep13 meant. In the case where a non-exempt dependency fails, it would be nice if a differentiation from a main package failure could be made. Maybe an additional verification error code (dependency failure) could be introduced which is also communicated differently on the package's gallery page.

**Comment by ferventcoder** [comment]
Hmm perhaps - that starts to complicate things a bit though - the verifier is really about pass fail and not really into scenarios.

**Comment by vktr** [comment]
I'm having the same issue, although I don't have a choco dependency on `vcredist2015` - the installer bundle will download and install it from Microsoft if it detects it missing.

**Comment by vktr** [comment]
Hi! What's the status for this one? Soon trusted packages will not be approved automatically even though the verifier fails.

Could we perhaps run the verifier on an updated version of Windows? The MSVC installer bug is fixed in KB2919355.

**Comment by ferventcoder** [comment]
Sounds like we need a package for KB2919355 that the redists can depend on.

**Comment by ferventcoder** [comment]
@vktr we use exemptions for now. If you need an exemption, you may want to use contact site admins for those packages.
